File: examples/algorithm.py
"""
Example: humanizing AI-flavored algorithm code (code-humanizer skill).

BEFORE = what an LLM typically emits.
AFTER  = the humanized version.
Both produce identical outputs on the normal path (verified by the asserts below).
"""

# ---------------- BEFORE (AI-sounding) ----------------
def calculate_power(base, exp):
    result = 1
    for i in range(exp):  # iterate exp times
        result = result * base
    return result

def remove_duplicates(arr):
    result = []
    for i in range(len(arr)):
        if arr[i] not in result:
            result.append(arr[i])
    return result

class SortUtility:
    def __init__(self):
        pass
    def sort_data(self, data):
        return sorted(data)

MOD = 1000000007

def solve(data):
    try:
        ans = 0
        for i in range(len(data)):
            ans = (ans + data[i]) % MOD
        return ans
    except Exception as e:
        print("An error occurred")
        return -1


# ---------------- AFTER (humanized) ----------------
def power(base, exp):
    return base ** exp

def unique(items):
    return list(dict.fromkeys(items))  # order-preserving, same behavior

MOD = 1_000_000_007

def sum_mod(values):
    return sum(values) % MOD


# ---------------- VERIFY (behavior preserved) ----------------
assert calculate_power(2, 10) == power(2, 10) == 1024
assert remove_duplicates([3, 1, 3, 2, 2]) == unique([3, 1, 3, 2, 2]) == [3, 1, 2]
assert solve([1, 2, 3]) == sum_mod([1, 2, 3]) == 6
assert solve([10**9, 10**9]) == sum_mod([10**9, 10**9]) == (2 * 10**9) % MOD
print("PASS: before and after match on all sample inputs")

File: examples/algorithm.go
// algorithm.go — cross-language before/after for code-humanizer (#16–#22).
// Run `go run algorithm.go` to confirm both versions agree on the sample input.

package main

import (
	"fmt"
	"math"
)

// ---------- BEFORE (AI-sounding) ----------

// this function calculates base raised to exp using a loop
func calculatePower(base float64, exp int) float64 {
	result := 1.0
	for i := 0; i < exp; i++ { // iterate exp times
		result = result * base
	}
	return result
}

// this function removes duplicate elements from a slice
func removeDuplicates(arr []int) []int {
	result := []int{}
	for i := 0; i < len(arr); i++ {
		found := false
		for j := 0; j < len(result); j++ {
			if result[j] == arr[i] {
				found = true
			}
		}
		if !found {
			result = append(result, arr[i])
		}
	}
	return result
}

func solveBefore(data []int) int {
	const MOD = 1000000007
	ans := 0
	for i := 0; i < len(data); i++ {
		ans = (ans + data[i]) % MOD
	}
	return ans
}

// ---------- AFTER (humanized) ----------

func power(base float64, exp int) float64 {
	return math.Pow(base, float64(exp))
}

func unique(items []int) []int {
	seen := make(map[int]struct{})
	out := make([]int, 0, len(items))
	for _, v := range items {
		if _, ok := seen[v]; ok {
			continue
		}
		seen[v] = struct{}{}
		out = append(out, v)
	}
	return out
}

const mod = 1_000_000_007

func sumMod(values []int) int {
	total := 0
	for _, v := range values {
		total = (total + v) % mod
	}
	return total
}

func main() {
	data := []int{3, 1, 3, 2, 2}

	// before
	fmt.Println(calculatePower(2, 10), removeDuplicates(data), solveBefore([]int{1e9, 1e9}))
	// after
	fmt.Println(power(2, 10), unique(data), sumMod([]int{1e9, 1e9}))
}
